daemon/smartcard.c

```diff
@@ -804,22 +801,36 @@ smartcard_cb_container_change_pin(int fd, unsigned events, event_io_t *io, void
 				      path, uuid_string(container_get_uuid(startdata->container)));
 				container_set_token_is_linked_to_device(startdata->container,
 									false);
+				audit_log_event(
+					container_get_uuid(startdata->container), FSA, CMLD,
+					CONTAINER_MGMT, "container-change-pin",
+					uuid_string(container_get_uuid(startdata->container)), 0);
 				control_send_message(CONTROL_RESPONSE_CONTAINER_CHANGE_PIN_FAILED,
 						     resp_fd);
 			} else {
 				container_set_token_is_linked_to_device(startdata->container, true);
+				audit_log_event(
+					container_get_uuid(startdata->container), SSA, CMLD,
+					CONTAINER_MGMT, "container-change-pin",
+					uuid_string(container_get_uuid(startdata->container)), 0);
 				control_send_message(
 					CONTROL_RESPONSE_CONTAINER_CHANGE_PIN_SUCCESSFUL, resp_fd);
 			}
 			mem_free(path);
 		} break;
 		case TOKEN_TO_DAEMON__CODE__PROVISION_PIN_FAILED: {
 			container_set_token_is_linked_to_device(startdata->container, false);
+			audit_log_event(container_get_uuid(startdata->container), FSA, CMLD,
+					CONTAINER_MGMT, "container-change-pin",
+					uuid_string(container_get_uuid(startdata->container)), 0);
 			control_send_message(CONTROL_RESPONSE_CONTAINER_CHANGE_PIN_FAILED, resp_fd);
 		} break;
 		default:
 			ERROR("TokenToDaemon command %d not expected as answer to change_pin",
 			      msg->code);
+			audit_log_event(container_get_uuid(startdata->container), FSA, CMLD,
+					CONTAINER_MGMT, "container-change-pin",
+					uuid_string(container_get_uuid(startdata->container)), 0);
 			control_send_message(CONTROL_RESPONSE_CONTAINER_CHANGE_PIN_FAILED, resp_fd);
 		}
```

Comment:
do log audit_log_evet and control_send_message accoriding to command_state here:
`audit_log_event(..., command_state ? SSA : FSA, ...)`
`control_send_message(command_state ? CONTROL_RESPONSE_CONTAINER_CHANGE_PIN_SUCCESSFUL : CONTROL_RESPONSE_CONTAINER_CHANGE_PIN_FAILED, resp_fd)`

Reply:
Done